java-shared-dependencies/dependency-analyzer/src/main/java/com/google/cloud/model/CheckReport.java

```diff
@@ -0,0 +1,92 @@
+package com.google.cloud.model;
+
+import com.google.cloud.exception.DependencyRiskException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class CheckReport {
+
+  private final VersionKey root;
+  private final Map<VersionKey, List<Advisory>> advisories;
+  private final Map<VersionKey, List<String>> nonCompliantLicenses;
+  private final List<LicenseCategory> nonCompliantCategories = List.of(LicenseCategory.Restricted);
```

Comment:
Would you use allowlist rather than denylist? It's more reliable to allow only certain values.

Reply:
Done.